azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub_mqtt.py

```diff
@@ -353,6 +390,40 @@ def test_sets_module_username_if_yes_gateway(self, stage, set_connection_args_fo
             encoded_user_agent,
         )
 
+    @pytest.mark.it(
+        "Appends product_info to connection_args.username to if self.pipeline_root.pipeline_configuration.product_info is not None"
+    )
+    @pytest.mark.parametrize(
+        "fake_product_info, expected_product_info",
+        [
+            ("", ""),
+            ("__fake_product_info__", "__fake_product_info__"),
```

Comment:
>("__fake_product_info__", "__fake_product_info__"), [](start = 12, length = 51)

What about just a string that needs encoding, like "fake,product,info", "fake%2Cproduct%2Cinfo"?    (I don't know why I'm so stuck on URI encoding).

Reply:
I can add another one. 

---
In reply to: [340334859](https://github.com/Azure/azure-iot-sdk-python/pull/368#discussion_r340334859) [](ancestors = 340334859)